File: src-tauri/src/tools/kv.rs
use reqwest::{Client, Method};
use serde::{Deserialize, Serialize};
use std::error::Error;

const KV_BASE_URL: &str = "https://kv.adswds.com/1ep2d8wb";

#[derive(Debug, Deserialize, Serialize)]
pub struct KvData {
    pub key: String,
    pub value: String,  // 这里直接是 String
}


// 响应体结构体
#[derive(Debug, Deserialize, Serialize)]
pub struct ApiResponse {
    pub code: u16,
    pub msg: String,
    pub data: Option<KvData>,
}

// 公共请求方法
async fn request(
    method: Method,
    authorization: &str,
    key: &str,
    value: Option<&str>,
) -> Result<ApiResponse, Box<dyn Error>> {
    let client = Client::new();

    let mut url = format!("{}?key={}", KV_BASE_URL, key);
    if let Some(v) = value {
        url = format!("{}&value={}", url, v);
    }

    let resp = client
        .request(method, &url)
        .header("Authorization", format!("Bearer {}", authorization))
        .send()
        .await?
        .json::<ApiResponse>()
        .await?;

    Ok(resp)
}

// 新增数据（POST）
pub async fn create_key(key: &str, value: &str, authorization: &str) -> ApiResponse {
    let res = request(Method::POST, authorization, key, Some(value)).await.unwrap();
    println!("POST result: {:?}", res);
    res
}

// 查询数据（GET）
pub async fn get_key(key: &str, authorization: &str) -> ApiResponse {
    let res = request(Method::GET, authorization, key, None).await.unwrap();
    println!("GET result: {:?}", res);
    res
}

// 更新数据（PUT）
pub async fn update_key(key: &str, value: &str, authorization: &str) -> ApiResponse {
    let res = request(Method::PUT, authorization, key, Some(value)).await.unwrap();
    println!("PUT result: {:?}", res);
    res
}

File: wails-version/adsplat/services/kv.go
package services

import (
	"encoding/json"
	"fmt"
	"io"
	"net/http"
	"net/url"
)

const KV_BASE_URL = "https://kv.adswds.com/1ep2d8wb"

// KvData KV存储数据结构
type KvData struct {
	Key   string `json:"key"`
	Value string `json:"value"`
}

// KvResponse KV API响应结构
type KvResponse struct {
	Code int     `json:"code"`
	Msg  string  `json:"msg"`
	Data *KvData `json:"data,omitempty"`
}

// KvService KV存储服务
type KvService struct {
	client *http.Client
}

// NewKvService 创建KV服务实例
func NewKvService() *KvService {
	return &KvService{
		client: &http.Client{},
	}
}

// request 通用请求方法
func (s *KvService) request(method, authorization, key, value string) (*KvResponse, error) {
	reqURL := fmt.Sprintf("%s?key=%s", KV_BASE_URL, url.QueryEscape(key))
	if value != "" {
		reqURL = fmt.Sprintf("%s&value=%s", reqURL, url.QueryEscape(value))
	}

	req, err := http.NewRequest(method, reqURL, nil)
	if err != nil {
		return nil, err
	}

	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", authorization))
	req.Header.Set("Content-Type", "application/json")

	resp, err := s.client.Do(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	body, err := io.ReadAll(resp.Body)
	if err != nil {
		return nil, err
	}

	var kvResp KvResponse
	if err := json.Unmarshal(body, &kvResp); err != nil {
		return nil, err
	}

	return &kvResp, nil
}

// CreateKey 创建键值对
func (s *KvService) CreateKey(key, value, authorization string) (*KvResponse, error) {
	return s.request("POST", authorization, key, value)
}

// GetKey 获取键值对
func (s *KvService) GetKey(key, authorization string) (*KvResponse, error) {
	return s.request("GET", authorization, key, "")
}

// UpdateKey 更新键值对
func (s *KvService) UpdateKey(key, value, authorization string) (*KvResponse, error) {
	return s.request("PUT", authorization, key, value)
}

// DeleteKey 删除键值对
func (s *KvService) DeleteKey(key, authorization string) (*KvResponse, error) {
	return s.request("DELETE", authorization, key, "")
}
